File: app/src/main/java/com/xuie/android/ui/adapter/MarginDecoration.java
package com.xuie.android.ui.adapter;

import android.content.res.Resources;
import android.graphics.Rect;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

/**
 * @author xuie
 */
public class MarginDecoration extends RecyclerView.ItemDecoration {
    private int margin;

    public MarginDecoration() {
        margin = (int) (8 * Resources.getSystem().getDisplayMetrics().density);
    }

    public MarginDecoration(int dp) {
        margin = (int) (dp * Resources.getSystem().getDisplayMetrics().density);
    }


    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.set(margin, margin, margin, margin);
    }
}

File: app/src/main/java/xuk/android/ui/adapter/MarginDecoration.kt
package xuk.android.ui.adapter

import android.content.res.Resources
import android.graphics.Rect
import android.view.View

import androidx.recyclerview.widget.RecyclerView

/**
 * @author Jie Xu
 */
class MarginDecoration : RecyclerView.ItemDecoration {
  private var margin: Int = 0

  constructor() {
    margin = (8 * Resources.getSystem().displayMetrics.density).toInt()
  }

  constructor(dp: Int) {
    margin = (dp * Resources.getSystem().displayMetrics.density).toInt()
  }


  override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
    outRect.set(margin, margin, margin, margin)
  }
}